runners/spark/3/src/main/java/org/apache/beam/runners/spark/structuredstreaming/translation/SparkSessionFactory.java

```diff
@@ -130,19 +170,35 @@ private static SparkSession.Builder sessionBuilder(
     // mode, so try to align with value of "sparkMaster" option in this case.
     // We should not overwrite this value (or any user-defined spark configuration value) if the
     // user has already configured it.
-    if (master != null
-        && !master.equals("local[*]")
-        && master.startsWith("local[")
-        && System.getProperty("spark.sql.shuffle.partitions") == null) {
-      int numPartitions =
-          Integer.parseInt(master.substring("local[".length(), master.length() - 1));
-      if (numPartitions > 0) {
-        sparkConf.set("spark.sql.shuffle.partitions", String.valueOf(numPartitions));
-      }
+    int partitions = localNumPartitions(master);
+    if (partitions > 0) {
+      sparkConf.setIfMissing("spark.sql.shuffle.partitions", Integer.toString(partitions));
     }
+
     return SparkSession.builder().config(sparkConf);
   }
 
+  @SuppressWarnings({"return", "toarray.nullable.elements", "methodref.receiver"}) // safe to ignore
+  private static String[] filesToStage(
+      SparkStructuredStreamingPipelineOptions opts, Collection<String> excludes) {
+    Collection<String> files = opts.getFilesToStage();
+    if (files == null || files.isEmpty()) {
+      return EMPTY_STRING_ARRAY;
+    }
+    if (!excludes.isEmpty()) {
+      files = Collections2.filter(files, f -> !excludes.stream().anyMatch(f::contains));
+    }
+    return files.toArray(EMPTY_STRING_ARRAY);
+  }
+
+  private static String[] sparkJars(SparkConf conf) {
```

Comment:
ping ?

Reply:
I prefer the name as is to be honest

Reply:
nit: I'd add a verb to the current name, like `retrieveSparkJars()` or similar, since it's a method, not variable nor constant.

Reply:
I'm sorry @aromanenko-dev, you got me started about this  😛 
I personally consider the "a method must contain a verb" convention out of time and even modern Java (fluent builders finally skipping the meaningless `with` prefix, record types getting rid of the `get` prefix) is slowly getting there. In my personal view a prefix like `get`, `retrieve`, `do`, `return` has close to no meaning (plus it doesn't really support understanding) and, consequently, is better skipped.  Luckily, the syntax makes it crystal clear that this is a method and not a variable nor constant, so we don't have to rely on placeholder verbs :) 
But fair argument below, that method does in deed filter as well - that is a difference worth mentioning using a verb in my eyes :)
But never mind, i can change to `getSparkJars` - not a big deal 😄

Reply:
I don't see what is a real problem to rename it (and actually name it initially) in a way how it got used to be on the project. Of course, we can find a bunch of exceptions if we want since Beam's code-base is quite large and developed by many different people in different time, but there are some code style rules that would be useful to follow since it's an open source project, not personal one. 

Anyway, it's not a big deal and it was just a recommendation from my side, I didn't insist.